v2/contribute/decisions/reactsdk/0002-split-recipe-route-handlers-into-separate-comps.mdx

```diff
@@ -0,0 +1,144 @@
+<ArgumentPro> Less interface change </ArgumentPro>
+<ArgumentCon> The recipes need to be listed in two places </ArgumentCon>
+<ArgumentCon> For most bundlers, this will import all recipes ui components instead of just the required ones </ArgumentCon>
+
+```tsx
+function App() {
+  return (
+    <SuperTokensWrapper>
+      <EmailVerificationComponentsOverrideProvider
+        components={{
+          EmailVerificationSendVerifyEmail_Override: () => <OtpScreen />,
+        }}>
+        <div className="App">
+          <Router>
+            <Routes>
+              {getSuperTokensRoutesForReactRouterDom(require("react-router-dom"), [EmailPassword.recipeId])}
+              <Route
+                path="/"
+                element={
+                  <SessionAuth>
+                    <Home />
+                  </SessionAuth>
+                }
+              />
+            </Routes>
+          </Router>
+        </div>
+      </EmailVerificationComponentsOverrideProvider>
+    </SuperTokensWrapper>
+  );
+}
+```
+
+### Add separate components for the pre-built UI of each recipe
+
+<ArgumentPro> Should work with all bundlers </ArgumentPro>
+<ArgumentPro> Interface feels native to react </ArgumentPro>
+<ArgumentPro> Adding a `componentOverrides` prop would replace the separate override provider component in most (but not all) cases </ArgumentPro>
+<ArgumentCon> The interface changes a lot </ArgumentCon>
+<ArgumentCon> The recipes have to be listed in two places </ArgumentCon>
+<ArgumentCon> `canHandleRoute` and `getRoutingComponent` will have to be recipe specific as well </ArgumentCon>
+
+With react-router-dom:
+
+```tsx
+function App() {
+  return (
+    <SuperTokensWrapper>
+      <EmailVerificationComponentsOverrideProvider
+        components={{
+          EmailVerificationSendVerifyEmail_Override: () => <OtpScreen />,
+        }}>
+        <div className="App">
+          <Router>
+            <Routes>
+              <EmailPasswordPreBuiltUIRoutes router={require("react-router-dom")} />
+              <EmailVerificationPreBuiltUIRoutes router={require("react-router-dom")} />
+              <Route
+                path="/"
+                element={
+                  <SessionAuth>
+                    <Home />
+                  </SessionAuth>
+                }
+              />
+            </Routes>
+          </Router>
+        </div>
+      </EmailVerificationComponentsOverrideProvider>
+    </SuperTokensWrapper>
+  );
+}
+```
+
+With no react-router-dom:
+
+```tsx
+class App extends React.Component {
+  render() {
+    // We could handle looping through these for the user but I don't think this is an issue.
+    if (EmailPasswordPreBuiltUI.canHandleRoute()) {
+      return EmailPasswordPreBuiltUI.getRoutingComponent();
+    }
+
+    if (EmailVerificationPreBuiltUI.canHandleRoute()) {
+      return EmailVerificationPreBuiltUI.getRoutingComponent({
+        EmailVerificationSendVerifyEmail_Override: () => <OtpScreen />,
```

Comment:
Would this have component override? Cause i thought we would be using `EmailVerificationComponentsOverrideProvider`, in which case, how would component override work here exactly?